pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -476,7 +476,7 @@ func (c *MaxPDVolumeCountChecker) predicate(pod *v1.Pod, meta Metadata, nodeInfo
 	maxAttachLimit := c.maxVolumeFunc(node)
 
 	volumeLimits := nodeInfo.VolumeLimits()
-	if maxAttachLimitFromAllocatable, ok := volumeLimits[c.volumeLimitKey]; ok {
+	if maxAttachLimitFromAllocatable, ok := volumeLimits[c.volumeLimitKey]; ok && getMaxVolLimitFromEnv() < 0 {
```

Comment:
Looks fine, it would only be overwritten if the environment variable is not set right?

Reply:
Yes, when the env-var is not set, maxAttachLimit will be overwritten by the value found in the NodeAllocatable.

Reply:
@hardikdr, I think you have to submit the PR against the master branch and once it's merged there, backport the patch all the way down to v1.17 (you may check the guidelines to be sure).

Also, it seems like the limit from the Node always had precedence over the limit from the env var. I don't know if this is an oversight or the intended behaviour. I'll try to find this out and keep you in the loop.

In addition to that, we are going to need unit tests for this change if we decide to go with it.

Reply:
@bertinatto Thanks for looking into it. 
I think this change is only applicable on 1.17,  the code-path is completely changed starting from 1.18. There, in 1.18, also we need to fix the working of the env-var, but I don't think it'll exactly be the same change as here or even this trivial. 

>Also, it seems like the limit from the Node always had precedence over the limit from the env var.
I think the env-var should have the highest priority, it's purpose is to overwrite existing values. My best guess would be, it's an oversight :)

I'll add the unit-tests, thanks.

Reply:
@hardikdr 

> @bertinatto Thanks for looking into it.
> I think this change is only applicable on 1.17, the code-path is completely changed starting from 1.18. There, in 1.18, also we need to fix the working of the env-var, but I don't think it'll exactly be the same change as here or even this trivial.

I'm not sure about the guidelines, but I think we need to change the code in master first anyway, then cherry-pick the change to older releases. It won't be an automated cherry-pick for sure.

The predicates code migrated to scheduler plugins in v1.18, however, the code we are interested is the same.

In v1.17: https://github.com/kubernetes/kubernetes/blob/d1c2f63bd4fc89d80e9bb6d21b5c1c24d241ad25/pkg/scheduler/algorithm/predicates/predicates.go#L475-L481
In master:  https://github.com/kubernetes/kubernetes/blob/633ab1ca61981608ffd0cfee61d40d0b6399680f/pkg/scheduler/framework/plugins/nodevolumelimits/non_csi.go#L250-L255

Reply:
Perfect, thanks I'll also open a PR on master then.